apps/meteor/client/views/omnichannel/directory/chats/contextualBar/RoomEdit.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { ReactElement, useCallback } from 'react';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+
+import CustomFieldsForm from '../../../../../components/Omnichannel/CustomFieldsForm';
+import Tags from '../../../../../components/Omnichannel/Tags';
+import VerticalBar from '../../../../../components/VerticalBar';
+import { useIsEnterprise } from '../../../../../hooks/useIsEnterprise';
+import { FormSkeleton } from '../../Skeleton';
+import { usePriorities } from './hooks/usePriorities';
+
+const CUSTOM_FIELDS_PERMISSION = ['view-livechat-room-customfields', 'edit-livechat-room-customfields'];
+const DEFAULT_VALUES = {
+	topic: '',
+	tags: [],
+	livechatData: {},
+	priorityId: '',
+};
+
+type RoomEditSaveInfo = {
+	topic: IOmnichannelRoom['topic'];
+	tags: IOmnichannelRoom['tags'];
+	livechatData: IOmnichannelRoom['livechatData'];
+	priorityId: IOmnichannelRoom['priorityId'];
+};
+
+const getInitialValuesRoom = (room: IOmnichannelRoom): RoomEditSaveInfo => {
+	if (!room) {
+		return DEFAULT_VALUES;
+	}
+
+	return {
+		topic: room.topic ?? '',
+		tags: room.tags ?? [],
+		livechatData: room.livechatData ?? {},
+		priorityId: room.priorityId ?? '',
+	};
+};
+
+const RoomEdit = ({
+	room,
+	visitor,
+	reload,
+	reloadInfo,
+	close,
+}: {
+	room: IOmnichannelRoom;
+	visitor: { _id: string };
+	reload: () => void;
+	reloadInfo: () => void;
+	close: () => void;
+}): ReactElement => {
+	const t = useTranslation();
+
+	const isEnterprise = useIsEnterprise();
+	const viewCustomFields = useAtLeastOnePermission(CUSTOM_FIELDS_PERMISSION);
+	const dispatchToastMessage = useToastMessageDispatch();
+	const saveRoom = useMethod('livechat:saveInfo');
+
+	const methods = useForm({ defaultValues: getInitialValuesRoom(room), mode: 'onChange' });
+
+	const {
+		control,
+		register,
+		getValues,
+		formState: { isDirty, isValid },
+	} = methods;
+
+	const { data: prioritiesData, isError: prioritiesError, isLoading: prioritiesLoading } = usePriorities();
+
+	const handleSubmit = useCallback(async () => {
+		const values = getValues();
+		const guestData = { _id: visitor._id };
+
+		const roomData = {
+			_id: room._id,
+			topic: values.topic,
+			tags: values.tags.sort(),
+			livechatData: values.livechatData,
+			...(values.priorityId && { priorityId: values.priorityId }),
+		};
+
+		try {
+			saveRoom(guestData, roomData);
+
+			dispatchToastMessage({ type: 'success', message: t('Saved') });
+
+			reload?.();
+			reloadInfo?.();
+
+			close();
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		}
+	}, [close, dispatchToastMessage, getValues, reload, reloadInfo, room._id, saveRoom, t, visitor._id]);
+
+	if (!prioritiesData || prioritiesLoading || prioritiesError) {
+		return <FormSkeleton />;
+	}
+
+	return (
+		<FormProvider {...methods}>
+			<VerticalBar.ScrollableContent is='form'>
```

Comment:
Usually when working with `react-hook-forms` you add a `onSubmit={handleSubmit(yourOnSaveFnHere)}`, and at the end of the form you add a button `type=submit`.